x-pack/plugins/security_solution/public/flyout/shared/components/flyout_navigation.tsx

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { FC } from 'react';
+import React, { memo } from 'react';
+import { EuiFlyoutHeader, EuiFlexGroup, EuiFlexItem, useEuiTheme } from '@elastic/eui';
+import { css } from '@emotion/react';
+import { ExpandDetailButton } from './expand_detail_button';
+import { HEADER_ACTIONS_TEST_ID } from './test_ids';
+
+export interface PanelNavigationProps {
+  /**
+   * If true, the expand detail button will be displayed
+   */
+  flyoutIsExpandable: boolean;
+  /**
+   * Expand left panel call back
+   */
+  expandDetails?: () => void;
+  /**
+   * Optional actions to be placed on the right hand side of navigation
+   */
+  actions?: React.ReactElement;
+}
+
+export const FlyoutNavigation: FC<PanelNavigationProps> = memo(
+  ({ flyoutIsExpandable = false, expandDetails, actions }) => {
+    const { euiTheme } = useEuiTheme();
+
+    return flyoutIsExpandable || actions ? (
+      <EuiFlyoutHeader hasBorder>
+        <EuiFlexGroup
```

Comment:
let's add `responsive={false}` here to avoid having things move around on small screens (see video below).

Also in the video below you'll see that the height of the component changes slightly depending on if the expand/collapse button is there or not. I think it would be nice to have the height fixed?


https://github.com/elastic/kibana/assets/17276605/2179f56e-ec34-42ef-b142-22a26a75b652

Reply:
great catch!